include/sys/vm_amap.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _SYS_VM_AMAP_H_
+#define _SYS_VM_AMAP_H_
+
+#include <sys/types.h>
+#include <sys/vm.h>
+#include <stddef.h>
+
+typedef struct vm_amap vm_amap_t;
+typedef struct vm_aref vm_aref_t;
+
+struct vm_aref {
+  size_t offset;   /* offset in slots */
+  vm_amap_t *amap; /* underlying amap */
+};
+
+#define VM_AREF_EMPTY ((vm_aref_t){.offset = 0, .amap = NULL})
+
+/*
+ * Allocate new amap with specified number of slots.
+ *
+ * Returns new amap with ref_cnt equal to 1.
+ */
+vm_amap_t *vm_amap_alloc(size_t slots);
```

Comment:
I think it always returns non-NULL value. Please verify and remove NULL checks.

Reply:
Yes. According to previous comments we will always get requested memory and return with valid amap.